### PR TITLE
Fit cube using parallelization

### PR DESCRIPTION
**Pull request by ibusko**
The relevant notebook, `Fitting_multiprocess.ipynb`, contains a proof-of concept for performing spectral fits on a JWST data cube, using multiprocessing to parallelize over multiple spaxels.

This is the first step for providing parallel cube fitting in Cubeviz.

### CODE REVIEW

File: notebooks/concepts/Fitting_multiprocess.ipynb [hunk 1]
```diff
@@ -0,0 +1,1742 @@
...
+    "\n",
+    "In here, we use the code from `Example_fitting.ipynb`. We disregard the error array for now, since the purpose is not to measure the input data itself, but instead to measure performance."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Define spectrum at the central spaxel"
+   ]
```
Comment (PatrickOgle) [inline notebooks/concepts/Fitting_multiprocess.ipynb:160]: This is the right idea--fit a representative spectrum to get starting values for the fits of all spaxels. The viz tool should have the flexibility to do this either on a user-specified spaxel or another spectrum, for example the average spectrum of the cube.  The advantage of the latter is that not all spaxels will contain all spectral features of interest, and also that the user doesn't have to think about which spaxel is best.

File: notebooks/concepts/Fitting_multiprocess.ipynb [hunk 2]
```diff
@@ -0,0 +1,1742 @@
...
+   "metadata": {},
+   "source": [
+    "There seems to be no penalty in calling the `Spectrum1D` constructor for every spaxel. The difference in execution times between the two approaches is of a few percent for cubes with up to 2500 spaxels.\n",
+    "\n",
+    "This is actually important to know, for the reason that follows:\n",
+    "\n",
+    "It hapens that passing a `Spectrum1D` instance to a multiprocess pool worker, prevents the worker from being called by the pool. The spectrum instance does not have even to participate in the actual execution, just a reference to it, if stored in the worker instance, somehow inhibits the worker from being called by the pool.\n",
+    "\n",
+    "This was tested with the worker object as an instance of a class, as below, and also with the worker being a plain top level function to which the spectrum gets passed at every call. No difference. "
+   ]
```
Comment (PatrickOgle) [inline notebooks/concepts/Fitting_multiprocess.ipynb:1204]: So you can't pass a Spectrum1D to a pool worker?  I ran into a similar problem with passing an lmfit method to multiprocessing pool.  My solution was to wrap it in a higher-level function.

File: notebooks/concepts/Fitting_multiprocess.ipynb [hunk 3]
```diff
@@ -0,0 +1,1742 @@
...
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Total time:  32.770500898361206 s\n",
+      "Time per spaxel: 0.03410041716790969 s\n"
+     ]
```
Comment (PatrickOgle) [inline notebooks/concepts/Fitting_multiprocess.ipynb:1332]: It looks like there was only a factor of 3 speedup compared to serial processing.  How many cores were used?

File: notebooks/concepts/Fitting_multiprocess.ipynb [hunk 4]
```diff
@@ -0,0 +1,1742 @@
...
+   "source": [
+    "#### Set up callback to collect results"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "2-D fit results (model parameters) come with different units associated to each 2-D array. So instead of a cube with fit results, we build a list of 2-D arrays storing `Quantity` instances.\n",
+    "\n",
```
Comment (PatrickOgle) [inline notebooks/concepts/Fitting_multiprocess.ipynb:1289]: I don't see why we can't construct a single model cube, with units attached to each parameter.

File: notebooks/concepts/Fitting_multiprocess.ipynb (hunk 2 above)
Comment (ibusko) [reply notebooks/concepts/Fitting_multiprocess.ipynb:1204]: I tried several different methods for passing a spectrum to the pool, but nothing worked. The only way that it worked for me is the one in the notebook, namely, pass a bare array and instantiate a Spectrum1D in the worker function.  Fortunately that doesn't seem to be a costly solution in terms of execution speed. I propose we do the implementation in this way and tackle the pool mysterious behavior in a separate issue.

File: notebooks/concepts/Fitting_multiprocess.ipynb (hunk 3 above)
Comment (ibusko) [reply notebooks/concepts/Fitting_multiprocess.ipynb:1332]: I tried with 8 and 7 cores.

File: notebooks/concepts/Fitting_multiprocess.ipynb (hunk 4 above)
Comment (ibusko) [reply notebooks/concepts/Fitting_multiprocess.ipynb:1289]: When we combine Quantity arrays with different units in each, into a higher dimension array, they seem to loose their units attributes and revert back to arrays:

**Comment by PatrickOgle** [comment]
@nmearl Why are checks failing for this PR, which is otherwise ready to merge?

**Pull request closed without merging**